### PR TITLE
Use generic class for load series

### PR DESCRIPTION
**Pull request by payetvin**
#1669

### CODE REVIEW

File: src/libs/antares/study/parts/load/container.cpp [hunk 1]
```diff
@@ -34,46 +34,40 @@ using namespace Yuni;
...
-    return sizeof(Container) + ((!series) ? 0 : DataSeriesLoadMemoryUsage(series))
-           + ((!prepro) ? 0 : prepro->memoryUsage());
+    return sizeof(Container) + series.memoryUsage() + ((!prepro) ? 0 : prepro->memoryUsage());
 }
 
 void Container::resetToDefault()
 {
-    if (series)
-        series->timeSeries.reset(1, HOURS_PER_YEAR);
+    series.reset();
```
Comment (flomnes) [inline src/libs/antares/study/parts/load/container.cpp:70]: Keep arguments ?

File: src/libs/antares/study/parts/load/container.h [hunk 2]
```diff
@@ -27,7 +27,7 @@
 #ifndef __ANTARES_LIBS_STUDY_PARTS_LOAD_CONTAINER_H__
 #define __ANTARES_LIBS_STUDY_PARTS_LOAD_CONTAINER_H__
 
-#include "series.h"
+#include <antares/study/parts/common/series.h>
```
Comment (flomnes) [inline src/libs/antares/study/parts/load/container.h:30]: How about
```suggestion
#include <antares/study/series.h>
```
?

File: src/solver/simulation/common-hydro-remix.cpp [hunk 3]
```diff
@@ -96,8 +96,8 @@ static bool Remix(const Data::AreaList& areas, PROBLEME_HEBDO& problem, uint num
             double bottom = std::numeric_limits<double>::max();
             double top = 0;
 
-            uint loadTS = area.load.series->getIndex(problem.year);
-            auto& load = area.load.series->timeSeries;
+            uint loadTS = area.load.series.getSeriesIndex(problem.year);
+            auto& load = area.load.series.timeSeries;
```
Comment (flomnes) [inline src/solver/simulation/common-hydro-remix.cpp:100]: We could use `getColumn` here

File: src/libs/antares/study/parts/load/container.h (hunk 2 above)
Comment (payetvin) [reply src/libs/antares/study/parts/load/container.h:30]: doesn't work

**Pull request merged**